status-im/status-react
# Add ios permissions checking on transition

fixes #1853 

### Summary:
The ios and android screens for send transaction have slightly different flows when requesting QR permissions. The android app requests permission on navigating to the send screen to activate QR scanning and the ios app presents a button that allows the user to get permissions. This pull request implements the ios style of flow, presenting an button that allows the user to grant permissions to the screen.

The screen before permissions have been secured:
<img width="370" alt="screen shot 2017-09-20 at 11 49 51 am" src="https://user-images.githubusercontent.com/640347/30650678-e8ed387a-9df9-11e7-870c-33923395052d.png">

And after:
<img width="364" alt="screen shot 2017-09-20 at 11 49 59 am" src="https://user-images.githubusercontent.com/640347/30650686-ec3e13b4-9df9-11e7-9cdf-0f018ce32903.png">


### Steps to test:
- On an ios device, reset permissions
- Navigate to send transaction screen
- Confirm that button requesting permissions exists


[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready
